### PR TITLE
fix(components): make inputs.model_artifact_url optional in sagemaker model component

### PR DESCRIPTION
**Pull request by ittus**
**Description of your changes:**
- It's ok to skip model_data_url when creating Sagemaker Model if the docker container already contains model artifact

```python
container = {"Image": docker_image }

create_model_response = sm.create_model(
    ModelName=model_name, ExecutionRoleArn=role, PrimaryContainer=container
)
```

This PR make `inputs.model_artifact_url` optional

### CODE REVIEW

**Review by akartsky — APPROVED** [review]

**Comment by akartsky** [comment]
Thank you

**Comment by surajkota** [comment]
@ittus thanks, can you also update the Notes section for runtime arguments in [this readme](https://github.com/kubeflow/pipelines/tree/master/components/aws/sagemaker/model#runtime-arguments). We want to make a few more changes(see below), is it ok if we coordinate it via this PR?

@ananth102 can you pick up changes from [this PR](https://github.com/kubeflow/pipelines/pull/7258) and create a PR to Thang's branch (https://github.com/ittus/pipelines/tree/fix/sagemaker-model-component) so we can release both these fixes in one release? Please help coordinate the release process

**Comment by ittus** [comment]
@ananth102 
I've cherry-picked from your branch. Could you please help to re-check?
I noticed that the image name is `abashyam123/kfp10:2` now.

**Comment by ittus** [comment]
@surajkota 
Thank you. [README.md](https://github.com/kubeflow/pipelines/pull/8336/commits/8029c64769c49156b861e0ade94e06bca546fa98) is updated.

**Comment by ananth102** [comment]
Hi @ittus thanks for all your help so far, can you cherry pick 9c0cb1e6c295d6ab4ed187740ebe63fb97ec906e (updated image and license
) from from this [branch](https://github.com/ananth102/pipelines/commits/kfpv1fixes2).

**Comment by ittus** [comment]
@ananth102 
I've cherry-picked it.

Is it possible to include this PR https://github.com/kubeflow/pipelines/pull/8335 in this release?

**Comment by ananth102** [comment]
Yea that can be done. Squash and cherry pick your changes from #8335.

**Comment by ittus** [comment]
@ananth102 
I've cherry-picked the PR and also made a small change https://github.com/kubeflow/pipelines/pull/8336/commits/a79fe7d365080db9fab552299ba51e036b5ba1bf

**Comment by ananth102** [comment]
Thanks for that small change! Once 41729d711cdacd5f6be48ce175e419dcc6c91771 is cherry picked, the team can cut a release.

**Comment by ittus** [comment]
> Once https://github.com/kubeflow/pipelines/commit/41729d711cdacd5f6be48ce175e419dcc6c91771 is cherry picked,

I've cherry-picked. Thanks!

File: components/aws/sagemaker/batch_transform/src/sagemaker_transform_component.py [hunk 1]
```diff
@@ -127,9 +139,19 @@ def _create_job_request(
...
-        request["DataProcessing"]["OutputFilter"] = inputs.output_filter
+
+        if inputs.resource_encryption_key:
+            request["TransformResources"][
+                "VolumeKmsKeyId"
+            ] = inputs.resource_encryption_key
+        else:
+            request["TransformResources"].pop("VolumeKmsKeyId")
+
+        if inputs.input_filter:
```
Comment (surajkota) [inline components/aws/sagemaker/batch_transform/src/sagemaker_transform_component.py:150]: Looks like all the parameters of data processing are being marked as optional. input_filter, output_filter are now optional and join_source was already optional

if none of the parameters for [data processing](https://github.com/kubeflow/pipelines/blob/master/components/aws/sagemaker/common/templates/transform.template.yaml#L24) are provided, it should be popped?

Can you please add this corresponding logic?

**Review by surajkota — COMMENTED** [review]
One comment

File: components/aws/sagemaker/batch_transform/src/sagemaker_transform_component.py (hunk 1 above)
Comment (ittus) [reply components/aws/sagemaker/batch_transform/src/sagemaker_transform_component.py:150]: Thanks for your review.
I added the logic to check https://github.com/kubeflow/pipelines/pull/8336/commits/1c014590377e1569d029770364ade1d9fb840ecc

Comment (surajkota) [reply components/aws/sagemaker/batch_transform/src/sagemaker_transform_component.py:150]: Hey @ittus, by default the [request[DataProcessing] will not be empty](https://github.com/kubeflow/pipelines/blob/master/components/aws/sagemaker/common/templates/transform.template.yaml#L24-L27) so the following if condition will never be true
```
if not request["DataProcessing"]:
     request.pop("DataProcessing")
```

I made some assumptions in my previous comment based on the fields but after reading [this documentation](https://docs.aws.amazon.com/sagemaker/latest/dg/batch-transform-data-processing.html#batch-transform-data-processing-workflow) I realized that we only need to pop the filters if not specified.

We just need to do:
```
if inputs.output_filter:
    request["DataProcessing"]["OutputFilter"] = inputs.output_filter
else:
    request["DataProcessing"].pop("OutputFilter")
```
same for input filter

**Review by surajkota — CHANGES_REQUESTED** [review]
we should be ready to merge with this change. Everything else looks good

File: components/aws/sagemaker/batch_transform/src/sagemaker_transform_component.py (hunk 1 above)
Comment (ittus) [reply components/aws/sagemaker/batch_transform/src/sagemaker_transform_component.py:150]: @surajkota 
Thanks for checking again.
I fixed in https://github.com/kubeflow/pipelines/pull/8336/commits/aa8764ddf598f31a9a5ffcaceb58c3e9247e192b
Please help to recheck 🙇‍♂️

**Review by surajkota — APPROVED** [review]
/approve

cc @ananth102

**Comment by ananth102** [comment]
This PR looks good to me, when I did some manual testing I found an unrelated url path bug for the logs. I've patched it cc82b67ae58fe2e9a99dfb177d1206e3c68a8a19 in [optfixtst](https://github.com/ananth102/pipelines/compare/optfixtst). I've run the integration test on the image that has the fix and did a manual test, both pass. Once cc82b67ae58fe2e9a99dfb177d1206e3c68a8a19 we can do the release as all the reviewers have approved the PR.

**Comment by ananth102** [comment]
Thanks for your contribution!

**Pull request merged**